### ISSUE TITLE
Custom SSO provider

### ISSUE THREAD

**Issue by mikemadisonweb** [Flagsmith/flagsmith#49]
Hello and thanks for an awesome open-source product!
I have a question. I saw Google OAuth integration added recently, is there a possibility to write your own authentication provider and configure Bullet train(on-premise) to use it? We have an internal SSO solution based on OAuth. If there is an already implemented functionality for LDAP, it could also be a solution for us.

**Comment by dabeeeenster** [comment]
Hi @mikemadisonweb - thanks for the kind words! We have some work done around SAML auth but nothing based around LDAP. Are you able to describe what you need to integrate with in more detail?

**Comment by mikemadisonweb** [comment]
What would be the best is an ability to set up a custom authentication provider. I'm not a Python pro, but we did something similar recently while integrating Sentry, an open-source error tracking tool, which is also a Django application. We have created a module with the provider inside of it, executed `pip install ./sentry-oauth` during the build, and Sentry used that provider for the authentication. I have put that module here for you to check: https://bitbucket.org/mikemadweb/sentry-oauth/src/master/
Is there a possibility to do something similar?
